src/Aspire.Hosting.Docker/DockerComposeServiceExtensions.cs

```diff
@@ -48,4 +48,60 @@ public static IResourceBuilder<T> PublishAsDockerComposeService<T>(this IResourc
 
         return builder;
     }
+
+    /// <summary>
+    /// Creates a placeholder for an environment variable in the Docker Compose file.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="dockerComposeService"></param>
+    /// <returns></returns>
+    public static string AsEnvPlaceHolder(this IManifestExpressionProvider builder, DockerComposeServiceResource dockerComposeService)
+    {
+
+        // Placeholder for resolving the actual parameter value
+        // https://docs.docker.com/compose/how-tos/environment-variables/variable-interpolation/#interpolation-syntax
+
+        // Treat secrets as environment variable placeholders as for now
+        // this doesn't handle generation of parameter values with defaults
+        var env = builder.ValueExpression.Replace("{", "")
+                 .Replace("}", "")
+                 .Replace(".", "_")
+                 .Replace("-", "_")
+                 .ToUpperInvariant();
+
+        dockerComposeService.Parent.AddEnvironmentVariable(env);
+
+        return $"${{{env}}}";
+    }
+
+    /// <summary>
+    /// Creates a placeholder for an environment variable in the Docker Compose file.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="dockerComposeService"></param>
+    /// <returns></returns>
+    public static string AsEnvPlaceHolder(this IResourceBuilder<ParameterResource> builder, DockerComposeServiceResource dockerComposeService)
```

Comment:
This appears unused unless I missed something?

Reply:
This is just a new public api. Mirrors AsProvisioningParameter